### DISCUSSION TITLE
Gentoo Prefix (with changable hardcoded path)

### DISCUSSION THREAD

**Discussion by littlewu2508** [termux/termux-packages#17478] [Ideas]
Hello, I'm a Gentoo perfix user on Android. I noticed that there is a [discussion about Gentoo](https://github.com/termux/termux-packages/issues/4031) previously, and states that Gentoo prefix is not suitable due to hardcoded path. However, I've found a way to overcome this: 

https://github.com/littlewu2508/migrate-prefix 

and

https://www.mail-archive.com/gentoo-alt@lists.gentoo.org/msg07044.html. 

They are basicly doing a text/binary replacement on hardcoded path, and the result seemed promising. There no failure of migration reported on the Gentoo base system.

There are pros of Gentoo prefix. It has its own libc, which means it only rely on kernel, and provide nearly vanilla Gentoo experience; it does not require chroot, so it can directly use and access files outside the prefix, e.g. calling binaries in `/system/bin`.

**Comment by littlewu2508** [comment]
@ulist @ghost You maybe interested in this

**Comment by littlewu2508** [comment]
And I am using Gentoo prefix on Android at `/data/gentoo64`. It's not broken

**Comment by littlewu2508** [comment]
I plan to raise up a proposal to Gentoo community, for making prefix stage3 tarballs as well as vanilla Gentoo stage3 tarballs. With that users can directly download, extract, and migrate the prefix to the location they like, without the procedure of bootstraping. That means we won't be limited to `/data/gentoo64`

**Comment by mio-19** [comment]
How do I install Gentoo Prefix on Android now? I found that pre-built tarballs were all removed :(

**Reply by littlewu2508** [reply]
You mean https://wiki.gentoo.org/wiki/Project:Android/tarball ? I'll ask it on ircs://irc.gentoo.org/#gentoo-prefix

I personally have rolled it to 2023 (when I raised this issue)

**Reply by littlewu2508** [reply]
Someone provide me a still workable mirror: https://mirror.litnet.lt/gentoo/experimental/prefix/rap/prefix-stage3-arm64-20200704.tar.xz

I also have one rolled to 20230825's portage tree at https://hep.tsinghua.edu.cn/~wuyy/gentoo64-20230825.tar.xz , sha256 `6ac7aee21a0bcf61c9de305d6f497c94cb4488eb2c2a8bc2f86ee3b34b0ea1a2`. You can try it if you trust me :)
